Feat/upgrade monorepo

adamtretera opened the pull request:
- Move modules to monorepo structure
- Added turbo repo 
- Added ts-reset
- T3 env



TODO:
- fix lint issue
- build typeerror
- fix using any useLocalizedResolver 
- add shared scripts (turbo lint, turbo format... ?)

adamtretera committed:
:sparkles: Coping files

adamtretera committed:
:sparkles: Move modules to monorepo

adamtretera committed:
:label: Added ts-reset

adamtretera committed:
:label: Added T3 env

adamtretera committed:
:bug: Import type

cermakjiri commented:
It's definitely a good idea to provide the description with some reasoning why you've added this and that and some overview what has been done, especially with this amount of changes.

adamtretera commented:
> It's definitely a good idea to provide the description with some reasoning why you've added this and that and some overview what has been done, especially with this amount of changes.

Yeah sorry I actually created a description somehow didnt update the MR

cermakjiri committed:
✨ Extend git ignores files

cermakjiri commented on template/package.json:
```diff
@@ -1,103 +1,25 @@
+        "prettier": "^3.0.3",
+        "turbo": "^1.10.14",
+        "typescript": "^5.2.2",
+        "@manypkg/cli": "^0.21.0"
```
It is used somewhere?

adamtretera replied:
added dependency check script - I think it pretty handy

cermakjiri commented on template/package.json:
```diff
@@ -1,103 +1,25 @@
-    "_id": "next-app-template@0.1.0"
-}
\ No newline at end of file
+    "prettier": "@workspace/prettier-config"
```
👍

cermakjiri commented on template/packages/ui/.svgrrc.js:
```diff
@@ -0,0 +1,42 @@
+/**
+ * @type {import('@svgr/core').Config}
+ */
+module.exports = {
```
👍

cermakjiri commented on template/packages/ui/package.json:
```diff
@@ -0,0 +1,62 @@
+        "storybook": "storybook dev -p 6006",
+        "build-storybook": "storybook build",
+        "format": "prettier --ignore-path .gitignore \"src/**/*.+(ts|js|tsx|json|html|css|md)\" --write",
+        "generate-icons": "npx @svgr/cli src/components/Icons/Icons/assets && yarn format && eslint --fix \"src/components/Icons/Icons/**/*.tsx\"",
```
I'll remove the npx when it's in the deps.

cermakjiri replied:
✅

Commit:
:wrench: Ignore all node_modules

Commit:
:wrench: Fix any cast in useLocalizedResolver

cermakjiri commented on template/tooling/eslint/tsconfig.json:
```diff
@@ -0,0 +1,8 @@
+    "compilerOptions": {
+        "tsBuildInfoFile": "node_modules/.cache/tsbuildinfo.json"
+    },
+    "include": ["."],
```
Shouldn't be the `node_modules/.cache/tsbuildinfo.json` part of `include`?

adamtretera replied:
No, its not a source code, so it shouldnt

cermakjiri commented on template/tooling/prettier/index.mjs:
```diff
@@ -0,0 +1,24 @@
+        "^(next/(.*)$)|^(next$)",
+        "<THIRD_PARTY_MODULES>",
+        "",
+        "^@workspace/(.*)$",
```
Nice naming 👍

cermakjiri commented on template/tooling/prettier/index.mjs:
```diff
@@ -0,0 +1,24 @@
+
+
+/** @typedef  {import("prettier").Config} PrettierConfig */
+/** @typedef  {import("@ianvs/prettier-plugin-sort-imports").PluginConfig} SortImportsConfig */
```
`PluginConfig` actually already include PrettierConfig

cermakjiri replied:
✅

cermakjiri commented on template/tooling/prettier/index.mjs:
```diff
@@ -0,0 +1,24 @@
+        "^[./]",
+    ],
+};
+export default  config
```
Looks great, I'll just use `generateImportOrder` from out prettier pkg to simplify it little bit.

cermakjiri replied:
✅

Commit:
:recycle: Wrap error package's code in src folder

Commit:
:bug: Export files from src instead of dist in ui package

cermakjiri committed:
🔧 Use turbo eslitnt config

cermakjiri committed:
⬆️ Install deps

cermakjiri committed:
♻️ Use generateImportOrder in prettier config

cermakjiri committed:
♻️ Don't use npx for turbo

It's installed as dep.

Commit:
:sparkles: Add logger package

Commit:
:wrench: Add more directories to gitignore

Commit:
:wrench: Add missing index.cjs file

cermakjiri committed:
♻️ Refactor intl module (+improve TS types)

cermakjiri commented on template/src/modules/logger/index.ts:
```diff
@@ -1,28 +0,0 @@
-    debug,
-    trace,
-    error,
-} as const;
```
This has been just deleted. Gonna create pkg for it.

cermakjiri replied:
✅

cermakjiri committed:
✨ Add logger pkg

cermakjiri committed:
♻️ Update tsconfig (remove redundant props)

cermakjiri committed:
🔧 Export from dist (the build dir), not from src in ui pkg

cermakjiri committed:
🚚 Don't wrap content of errors pkg to src dir

The errors pkg just exports common code that doesn't require translation or any build process.

cermakjiri committed:
♻️ Refactor deps & fix type errors

Leave dep. in packages, don't bring them to apps too if not necessary.
E.g.
`@workspace/tsconfig` includes `typescript` as its dep. Then we only need and have to use the `@workspace/tsconfig` and not to install the `typescript` separately  – it brings the `typescript` with itself.

cermakjiri committed:
♻️ Call skott with npx instead of having it installed as dep

Why? Because it has typescript@4.x dep which isn't compatible with our current stack on typescript@5.x and thus getting error during build that `verbatimModuleSyntax` isn't avail.

Commit:
:truck: Rename packages/localization/intl/index.tsx to index.ts

Commit:
:wrench: Add dist to gitignore

Commit:
Merge remote-tracking branch 'origin/feat/upgrade-monorepo' into feat/upgrade-monorepo

# Conflicts:
#	template/.gitignore
#	template/packages/localization/intl/index.tsx
#	template/packages/logger/index.ts
#	template/packages/logger/package.json

Commit:
:recycle: Rename packages/localization/intl/index.tsx to index.ts

cermakjiri committed:
🐛 Enforce prettier@3.0.3

cermakjiri commented:
- [x] fix lint issue – ✅ 
- [x] build typeerror – ✅ 
- [x] fix using any useLocalizedResolver – ✅  @2bad2furious  🙌
- [ ] add shared scripts (turbo lint, turbo format... ?) – ⏩  It'd be nice but we need the skeleton ASAP. I suggest to add it later based on the experience from the new projects.
- [ ] Lokse is working 🚧 


Also as we discussed on one the meeting sometime ago, it'd be nice to have `package/common` that includes all shared /  code among apps (`errors`, `localization`, etc.). This pkg doesn't require any build process, it's just plain TS files which are then directly imported to given app. However, I think we can skip for now and added later as well.

2bad2furious commented:
Does building the app work now? It failed prerendering home page due to some intl stuff before

cermakjiri commented:
> Does building the app work now? It failed prerendering home page due to some intl stuff before

That's has been fixed. However lokse doesn't work

Commit:
Merge remote-tracking branch 'origin/feat/upgrade-monorepo' into feat/upgrade-monorepo

adamtretera committed:
:sparkles: Added script for dependency check
